### PR TITLE
Added AsyncDaemonHealthCheck

### PR DESCRIPTION
**Pull request by daveHylde**
Added health check for the daemon

Note: Something went wrong when merging the latest oringin-changes to the fork so the original PR got deleted. :(

### CODE REVIEW

**Comment by daveHylde** [comment]
Btw - here's the original PR (with no files though):
https://github.com/JasperFx/marten/pull/2648

File: docs/events/projections/healthchecks.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Async Daemon HealthChecks
+
+Marten supports a customizable [HealthChecks](https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/health-checks?view=aspnetcore-7.0). This can be useful when running the async daemon in a containerized environment such as Kubernetes. The check will verify that no projection's progression lags more than `maxEventLag` behind the `HighWaterMark`. The default `maxEventLag` is 100.
```
Comment (oskardudycz) [inline docs/events/projections/healthchecks.md:3]: **SUGGESTION:** It'd be good to briefly explain what's max event lag.

Thank you for adding documentation :+1:

File: src/Marten/Events/Daemon/AsyncDaemonHealthCheckExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+using JasperFx.Core;
+using Marten.Events.Projections;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace Marten.Events.Daemon;
+
+public static class AsyncDaemonHealthCheckExtensions
+{
+    /// <summary>
```
Comment (oskardudycz) [inline src/Marten/Events/Daemon/AsyncDaemonHealthCheckExtensions.cs:15]: **COMMENT:** KUDOS for not forgetting about XML comments and adding code samples to them :+1:

File: docs/events/projections/healthchecks.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Async Daemon HealthChecks
```
Comment (oskardudycz) [inline docs/events/projections/healthchecks.md:1]: **FIX:** Sounds like our spell checker doesn't know the word "healthcheck" (see https://github.com/JasperFx/marten/actions/runs/5848701384/job/16131086583?pr=2679). Could you add it to the config: https://github.com/JasperFx/marten/blob/master/docs/cSpell.json#L4?

Comment (daveHylde) [reply docs/events/projections/healthchecks.md:1]: Sure - in the "words" section or "ignoreWords"?

File: src/Marten/Events/Daemon/AsyncDaemonHealthCheckExtensions.cs (hunk 2 above)
Comment (daveHylde) [reply src/Marten/Events/Daemon/AsyncDaemonHealthCheckExtensions.cs:15]: Thanks mate. Good XML comments is a must! :)

File: docs/events/projections/healthchecks.md (hunk 3 above)
Comment (oskardudycz) [reply docs/events/projections/healthchecks.md:1]: Yup 👍

**Comment by oskardudycz** [comment]
@daveHylde, I made a small tweak to docs and rebased with the main branch.

Thanks a lot for the work. Superb addition :+1:

**Pull request merged**